doc/subsystems/shell/shell.rst

```diff
@@ -0,0 +1,296 @@
+.. _shell:
+
+Shell
+######
+
+Overview
+********
+
+This module allows you to create and handle a shell with a user-defined command
+set. You can use it in examples where more than simple button or LED user
+interaction is required. This module is a Unix-like shell with these features:
+
+* Support for multiple instances.
+* Advanced cooperation with the :ref:`logger`.
+* Support for static and dynamic commands.
+* Smart command completion with the :kbd:`Tab` key.
+* Built-in commands: :command:`clear`, :command:`shell`, :command:`colors`,
+  :command:`echo`, :command:`history` and :command:`resize`.
+* Viewing recently executed commands using keys: :kbd:`↑` :kbd:`↓`.
```

Comment:
it would be worth to mention that some features are optional and can be turned of to reduce RAM or ROM.

Reply:
I've added information about Kconfig configuration.